recipe/meta.yaml

```diff
@@ -717,28 +727,125 @@ outputs:
+        - if not exist %SP_DIR%\pyarrow\arrow_python.dll exit 1                     # [win]
+        - if not exist %SP_DIR%\pyarrow\arrow_python_flight.dll exit 1              # [win]
+        - if not exist %SP_DIR%\pyarrow\arrow_python_parquet_encryption.dll exit 1  # [win]
+
+        - test -f ${SP_DIR}/pyarrow/include/arrow/python/pyarrow.h                  # [unix]
+        - if not exist %SP_DIR%\pyarrow\include\arrow\python\pyarrow.h exit 1       # [win]
+
+        - test ! -f ${SP_DIR}/pyarrow/tests/test_array.py                           # [unix]
+        - if exist %SP_DIR%/pyarrow/tests/test_array.py exit 1                      # [win]
         # Need to remove dot from PY_VER; %MYVAR:x=y% replaces "x" in %MYVAR% with "y"
         - if not exist %SP_DIR%/pyarrow/_cuda.cp%PY_VER:.=%-win_amd64.pyd exit 1  # [win and cuda_compiler_version != "None"]
+
+        # Expected not included libraries
+        - test ! -f $PREFIX/lib/libarrow_acero${SHLIB_EXT}      # [unix]
+        - test ! -f $PREFIX/lib/libarrow_dataset${SHLIB_EXT}    # [unix]
+        - test ! -f $PREFIX/lib/libarrow_flight${SHLIB_EXT}     # [unix]
+        - test ! -f $PREFIX/lib/libgandiva${SHLIB_EXT}          # [unix]
+        - test ! -f $PREFIX/lib/libparquet${SHLIB_EXT}          # [unix]
+
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: Python libraries for Apache Arrow Core
+
+  - name: pyarrow
+    version: {{ version }}
+    requirements:
+      host:
+        # only necessary for run-exports
+        - python
+        - numpy
+      run:
+        # Default set of libs because run-exports from libarrow-all aren't picked up
+        # Default doesn't contain flight, flight-sql and gandiva
+        - {{ pin_subpackage("libarrow", exact=True) }}
+        - {{ pin_subpackage("libarrow-acero", exact=True) }}
+        - {{ pin_subpackage("libarrow-dataset", exact=True) }}
+        - {{ pin_subpackage("libarrow-substrait", exact=True) }}
+        - {{ pin_subpackage("libparquet", exact=True) }}
+        - {{ pin_subpackage("pyarrow-core", exact=True) }}
+        - {{ pin_compatible('numpy') }}
+        - python
+      run_constrained:
+        - apache-arrow-proc =*={{ build_ext }}
+
+    test:
+      files:
+        - test_read_parquet.py
+      imports:
+        # pyarrow-all contains parquet
+        - pyarrow.dataset
+        - pyarrow.parquet
+      commands:
+        # Expected included libraries
+        - test -f $PREFIX/lib/libarrow_acero${SHLIB_EXT}      # [unix]
+        - test -f $PREFIX/lib/libarrow_dataset${SHLIB_EXT}    # [unix]
+        - test -f $PREFIX/lib/libparquet${SHLIB_EXT}          # [unix]
+
+        # Expected not included libraries
+        - test ! -f $PREFIX/lib/libarrow_flight${SHLIB_EXT}                        # [unix]
+        - test ! -f $PREFIX/lib/libgandiva${SHLIB_EXT}                             # [unix]
```

Comment:
This on the other hand is fine to keep it as a safety for ensuring we don't mess up our pyarrow layering (e.g. by adding the wrong dependencies).